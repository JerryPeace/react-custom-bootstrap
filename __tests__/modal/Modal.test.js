import React from 'react';
import Modal from 'modal';
import { mount } from 'enzyme';

describe('Modal', () => {
    let modal;
    let mockClose;
    let nodeProps;

    describe('with header/body/footer props', () => {
        beforeEach(() => {
            mockClose = jest.fn();
            nodeProps = {
                header: 'This is header.',
                body: 'This is body.',
                footer: 'This is footer.',
                onClose: mockClose
            };
            modal = mount(<Modal {...nodeProps} />);
        });

        test('should render corrently', () => {
            expect(modal).toMatchSnapshot();
        });

        test('click on close should hide modal', () => {
            modal.find('.close').simulate('click');
            expect(mockClose.mock.calls.length).toBe(1);
            expect(modal).toMatchSnapshot();
        });
    });

    describe('with show prop', () => {
        beforeEach(() => {
            nodeProps = {
                closeBtn: false,
                header: 'This is header.',
                body: 'This is body.',
                footer: 'This is footer.',
                show: true
            };
            modal = mount(<Modal {...nodeProps} />);
        });

        test('should render corrently', () => {
            expect(modal).toMatchSnapshot();
        });

        test('should hide after toggle show prop', () => {
            modal.setProps({show: false});
            expect(modal).toMatchSnapshot();
        });
    });
});
