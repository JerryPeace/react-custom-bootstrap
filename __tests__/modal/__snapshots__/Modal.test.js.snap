// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal with header/body/footer props click on close should hide modal 1`] = `
<Modal
  body="This is body."
  closeBtn={true}
  defaultShow={true}
  enableEscapeToClose={false}
  footer="This is footer."
  header="This is header."
  isLoading={false}
  onClose={[Function]}
>
  <ModalContainer
    bsClass="modal"
    bsSize="md"
    className=""
    defaultShow={true}
    isLoading={false}
    noCover={false}
    onEscapePressed={null}
    show={false}
  >
    <div
      className="modal"
      onKeyUp={null}
    >
      <div
        className="modal-dialog modal-md"
      >
        <div
          className="modal-content"
        >
          <Header
            bsClass="modal-header"
            closeBtn={true}
            onClick={[Function]}
          >
            <div
              className="modal-header"
            >
              <button
                aria-label="Close"
                className="close"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden={true}
                >
                  ×
                </span>
              </button>
              <h4
                className="modal-title"
              >
                This is header.
              </h4>
            </div>
          </Header>
          <Body
            bsClass="modal-body"
          >
            <div
              className="modal-body"
            >
              This is body.
            </div>
          </Body>
          <Footer
            bsClass="modal-footer"
          >
            <div
              className="modal-footer"
            >
              This is footer.
            </div>
          </Footer>
        </div>
      </div>
    </div>
  </ModalContainer>
</Modal>
`;

exports[`Modal with header/body/footer props should render corrently 1`] = `
<Modal
  body="This is body."
  closeBtn={true}
  defaultShow={true}
  enableEscapeToClose={false}
  footer="This is footer."
  header="This is header."
  isLoading={false}
  onClose={[Function]}
>
  <ModalContainer
    bsClass="modal"
    bsSize="md"
    className=""
    defaultShow={true}
    isLoading={false}
    noCover={false}
    onEscapePressed={null}
    show={true}
  >
    <div
      className="modal show"
      onKeyUp={null}
    >
      <div
        className="modal-dialog modal-md"
      >
        <div
          className="modal-content"
        >
          <Header
            bsClass="modal-header"
            closeBtn={true}
            onClick={[Function]}
          >
            <div
              className="modal-header"
            >
              <button
                aria-label="Close"
                className="close"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden={true}
                >
                  ×
                </span>
              </button>
              <h4
                className="modal-title"
              >
                This is header.
              </h4>
            </div>
          </Header>
          <Body
            bsClass="modal-body"
          >
            <div
              className="modal-body"
            >
              This is body.
            </div>
          </Body>
          <Footer
            bsClass="modal-footer"
          >
            <div
              className="modal-footer"
            >
              This is footer.
            </div>
          </Footer>
        </div>
      </div>
    </div>
  </ModalContainer>
</Modal>
`;

exports[`Modal with show prop should hide after toggle show prop 1`] = `
<Modal
  body="This is body."
  closeBtn={false}
  defaultShow={true}
  enableEscapeToClose={false}
  footer="This is footer."
  header="This is header."
  isLoading={false}
  show={false}
>
  <ModalContainer
    bsClass="modal"
    bsSize="md"
    className=""
    defaultShow={true}
    isLoading={false}
    noCover={false}
    onEscapePressed={null}
    show={false}
  >
    <div
      className="modal"
      onKeyUp={null}
    >
      <div
        className="modal-dialog modal-md"
      >
        <div
          className="modal-content"
        >
          <Header
            bsClass="modal-header"
            closeBtn={false}
            onClick={[Function]}
          >
            <div
              className="modal-header"
            >
              <h4
                className="modal-title"
              >
                This is header.
              </h4>
            </div>
          </Header>
          <Body
            bsClass="modal-body"
          >
            <div
              className="modal-body"
            >
              This is body.
            </div>
          </Body>
          <Footer
            bsClass="modal-footer"
          >
            <div
              className="modal-footer"
            >
              This is footer.
            </div>
          </Footer>
        </div>
      </div>
    </div>
  </ModalContainer>
</Modal>
`;

exports[`Modal with show prop should render corrently 1`] = `
<Modal
  body="This is body."
  closeBtn={false}
  defaultShow={true}
  enableEscapeToClose={false}
  footer="This is footer."
  header="This is header."
  isLoading={false}
  show={true}
>
  <ModalContainer
    bsClass="modal"
    bsSize="md"
    className=""
    defaultShow={true}
    isLoading={false}
    noCover={false}
    onEscapePressed={null}
    show={true}
  >
    <div
      className="modal show"
      onKeyUp={null}
    >
      <div
        className="modal-dialog modal-md"
      >
        <div
          className="modal-content"
        >
          <Header
            bsClass="modal-header"
            closeBtn={false}
            onClick={[Function]}
          >
            <div
              className="modal-header"
            >
              <h4
                className="modal-title"
              >
                This is header.
              </h4>
            </div>
          </Header>
          <Body
            bsClass="modal-body"
          >
            <div
              className="modal-body"
            >
              This is body.
            </div>
          </Body>
          <Footer
            bsClass="modal-footer"
          >
            <div
              className="modal-footer"
            >
              This is footer.
            </div>
          </Footer>
        </div>
      </div>
    </div>
  </ModalContainer>
</Modal>
`;
